genai-quickstart-pocs-python/amazon-bedrock-intelligent-document-processing-poc/pages/7_💬_Step_7:_Document_Q&A_with_Bedrock.py

```diff
@@ -0,0 +1,81 @@
+# Clear messages on screen if selectedoption changes
+if st.session_state.messages and st.session_state.option != option:
+    st.session_state.messages = []
+
+# Store current option in session state
+st.session_state.option = option
+
+# If user selected multimodal option, then list files from the S3 bucket above the chat history
+if option == 'Ask questions about the document using multimodal model':
+    # List files in S3 and allow user to select one
+    files = list_files_in_s3()
+    selected_file = st.selectbox("Select an image from S3 to process:", files)
+    if selected_file:
+         encoded_image = encode_image_to_base64(selected_file)
+
+# Display the messages stored in session state 
+for message in st.session_state.messages:
+    with st.chat_message(message["role"]):
+        st.markdown(message["content"])
+
+# If user selected text option, then answer questions from the local file
+if option == 'Ask questions about the document using contents from enriched_output.txt file':
+    # Read the enriched document text from file
+    with open("enriched_output.txt", "r", encoding="utf-8") as file:
+        file_text = file.read()
+    # Display chat input for user question
+    if question := st.chat_input("Ask a question about the document..."):
+        # Show the user's message
+        with st.chat_message("user"):
+            st.markdown(question)
+        # Store the user message in session state
+        st.session_state.messages.append({"role": "user", "content": question})
+        # Generate the response from Bedrock
+        with st.chat_message("assistant"):
+            message_placeholder = st.empty()
+            with st.status("Determining the best possible answer...", expanded=False) as status:
+                combined_prompt = f"\nDocument:\n{file_text}\n\nQuestion:\n{question}"
+                answer = chat_with_bedrock(combined_prompt)
+                message_placeholder.markdown(answer)  # Display the response
+                status.update(label="Question Answered...", state="complete", expanded=False)
+        # Store the assistant's response in session state
+        st.session_state.messages.append({"role": "assistant", "content": answer})
+
+# If user selected multimodal option, then answer questions from the S3 bucket
+elif option == 'Ask questions about the document using multimodal model':
+    # Display chat input for user question
+    if question := st.chat_input("Ask a question about the document..."):
+        # Show the user's message
+        with st.chat_message("user"):
+            st.markdown(question)
+        # Store the user message in session state
+        st.session_state.messages.append({"role": "user", "content": question})
+        # Generate the response from Bedrock
+        with st.chat_message("assistant"):
+            message_placeholder = st.empty()  # Placeholder for response
+            with st.status("Determining the best possible answer...", expanded=False) as status:
```

Comment:
Can you try swapping with a `st.spinner()`
The spinner is great because it shows when it's loading, but it disappears when the loading is complete.